### ISSUE TITLE
Composer fails to resolve dependency

### ISSUE THREAD

**Issue by maddes** [composer/composer#2617]
It doesn't detect that a previous version of a "wildcarded" package solves the dependency.

Example:

The main package is `'laravel/laravel'`
latest version is `v4.1.18`

which requires: `"laravel/framework": "4.1.*"`
which resolves currently to `v4.1.18`
which requires: `"patchwork/utf8": "1.1.*",`
which resolves as `v1.1.17`
which requires: `"lib-pcre": ">=7.9"`

Now, I've an old version of lib-pcre, so this all fails stating:

`patchwork/utf8 v1.1.17 requires lib-pcre >=7.9`

But, now as the dependency was `1.1.*`, it should check on other versions within that range that could meet the dependency, in fact `v1.1.16` does, as it requires `"lib-pcre": "*"` and solves the dependency...

In other words, it should resolve `patchwork/utf8 v1.1.16` instead of `patchwork/utf8 v1.1.17`.

Grapical explanation: (package: required => solved)

```
  "laravel/laravel" => "v4.1.18"
  -- "laravel/framework": "4.1.*" => "v4.1.18"
     -- "patchwork/utf8": "1.1.*" => "v1.1.17"
        -- "lib-pcre": ">=7.9" => "fail"
     -- "patchwork/utf8": "1.1.*" => "v1.1.16"
        -- "lib-pcre": "*" => "success (current as any will do)"
```
